lldb/source/Plugins/Process/elf-core/ThreadElfCore.h

```diff
@@ -145,6 +164,7 @@ struct ThreadData {
   int signo = 0;
   int code = 0;
   int prstatus_sig = 0;
+  std::string description;
```

Comment:
We might be able to remove all of these and just replace these with a `lldb::StopInfoSP` and always either create a `lldb::StopInfoSP` for the thread when we create the thread instead of duplicating info from the `siginfo_t` struct here, we don't want to keep adding more fields here in the future.

Reply:
I'm going to lightly push back on this. ProcessElfCore currently depends a lot on threaddata, not just for populating the thread but for some ELF specific logic. 

I think what we could do is forego having ELFSigInfo populate itself and instead directly populate threaddata, which prevents us from double allocating, then we can have `ThreadElfCore` continue creating a `StopInfoSP` in `CalculateStopInfo`.